### DISCUSSION TITLE
Trouble of downlaoding

### DISCUSSION THREAD

**Discussion by Yibh22** [neurodesk/neurodesk.github.io-hugo#399] [Q&A] [answered]
Dear all,

When trying to downlaod neurodesktop for Ubuntu 22.04, and running the script for step 1.b) without a persistent home directory,  I meet the following error:
CVMFS directory not there. Trying internal fuse mount next.
            ==================================================================
            Mounting CVMFS
autofs is NOT running - attempting to mount manually:
CernVM-FS: running with credentials 107:115
CernVM-FS: loading Fuse module... Failed to initialize root file catalog (16 - file catalog failure)

How to deal with it?

Thank you very much for your consideration and I am looking forward to your reply.

**Comment by stebo85** [comment]
Dear @Yibh22 - it looks like your firewall is blocking access to the CernVM-FS share. Can you try the following:

```
sudo docker run \
  --shm-size=1gb -it --privileged --user=root --name neurodesktop \
  -v ~/neurodesktop-storage:/neurodesktop-storage \
  -e NB_UID="$(id -u)" -e NB_GID="$(id -g)" \
  -e CVMFS_DISABLE=true \
  -p 8888:8888 \
  -e NEURODESKTOP_VERSION=2024-01-12 vnmd/neurodesktop:2024-01-12
```

If this is the core problem. Then it should start up ok, but now every container you use has to be downloaded to your filesystem and unpacked.

Another thing to try: Can you try to connect to a different network?

If you want to use CVMFS, you can also try installing CVMFS in your Ubuntu 22.04 host and see if it works there. If this solves your problem then it's not the firewall but the problem is related to your docker setup: https://www.neurodesk.org/docs/getting-started/neurocontainers/cvmfs/

Depending on the outcome of these we can try to narrow down the problem.

**Reply by Yibh22** [reply]
Dear @stebo85 , Thank you very much for your reply. I tried to install CVMFS on my host and configure it according to https://www.neurodesk.org/docs/getting-started/neurocontainers/cvmfs/. After doing this, I successfully downloaded Neurodesktop and opened it in the browser.

However, I am new to the neurodesk and I still have some question that confused me.

1. What is the role of the home directory? What is the difference between having a persistent home directory and not?

2. After running the docker-run script, there is some additional information different from the totorial.
[I 2024-01-31 06:40:59.287 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
Do I need to do something to install these non-installed servers? Will this affect my neuroimage processing?

3. What is the difference between neurodesk APP and this kind of browser-oped neurodesktop?

4. What is the difference between Neurodesktop, Neurocommond, and Neurocontainers? Just running the neurodesktop is enough to support my image processing, right? So there is no need to download Neurocommond and Neurocontainers?

Thank you very much for your consideration, and I am looking forward to your reply.

**Reply by stebo85** [reply]
see reply below

**Comment by Yibh22** [comment]
Dear @stebo85 , thank you very much for your reply. I tried to install CVMFS in my host and configure it according to https://www.neurodesk.org/docs/getting-started/neurocontainers/cvmfs/. After doing this, I successfully downloaded neurodesktop and open it in the browser. 

However, I am new to the neurodesk and I still have some question that confused me. 

1. What is the role of home directory? What is the difference between with and without a persistent home directory?

2. After running the docker-run script, there is some addition information different from the totorial. 
[I 2024-01-31 06:40:59.287 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
Do I need to do something to install these non-installed server? Will this affect my neuroimage processing?

3. What is the difference between neurodesk APP and this kind of browser-oped neurodesktop?

4. What is the difference between Neurodesktop, Neurocommond and Neurocontainers? Just running the neurodesktop is enough to support my image processing, right? so no need to download Neurocommond and Neurocontainers?

Thank you very much for your consideration and I am looking forward to your reply.

**Reply by stebo85** [reply, marked as answer]
Dear @Yibh22,

Wonderful to hear, it's working for you now :)

To answer your questions:
1) The homedirectory stores your settings in the Linux desktop. If you don't have a persistent homedirectory, restarting the docker container will always reset your settings. You can have a persistent homedirectory and that keeps your settings across updates and restarts.

2) you can ignore these warnings. You shouldn't need any of these.

3) Neurodeskapp controls the Docker Container for you. So if you don't want to start and stop the container or think about the persistent or non-persistant homedirectory or think about updating Neurodesktop: Use the app :) If you want to control these things yourself: Keep using the Docker command

4) Here you find all details: https://www.neurodesk.org/developers/architecture/neurodesk-components/ - in short: Neurodesktop is everything packaged up in a docker container. If you want only a command line interface and don't need a desktop you can use Neurocommand. If you only want a single container you can use Neurocontainers. Running Neurodesktop is all most people need and it brings everything else.
